harvester/harvest/alma.py

```diff
@@ -78,51 +89,94 @@ def parse_marc_records_from_files(self) -> Iterator[MARCRecord]:
+        )
+        if not extracted_dates:
+            raise AlmaCannotIdentifyLatestFullRunDateError
+        return extracted_dates[0]
 
-        Retrieve list of XML files from S3 or local filesystem, then yield filepaths that
-        match harvest options.
-        """
-        if self.input_files.startswith("s3://"):
-            filepaths = self._list_s3_xml_files()
-        else:
-            filepaths = self._list_local_xml_files()  # pragma: nocover
+    def _get_date_from_filepath(self, filepath: str) -> str:
+        """Get date string from filepath."""
+        match = FILEPATH_DATE_REGEX.match(filepath)
+        if not match:  # pragma: nocover
+            message = f"Could not parse date from filepath: {filepath}"
+            logger.warning(message)
+            return ""
+        return match.group(1)
 
-        for filepath in filepaths:
-            if not self._filter_filepath_by_harvest_type(filepath):
-                continue
-            if not self._filter_filepath_by_dates(filepath):
-                continue
-            yield filepath
+    def _filter_filepaths_by_harvest_type(self, filepaths: list[str]) -> list[str]:
+        """Filter list of XML files by harvest type.
 
-    def _filter_filepath_by_harvest_type(self, filepath: str) -> bool:
-        """Bool if harvest type aligns with record stage in filepath.
+        Given a list of XML files, the method will search for the presence of the
+        harvest type ("daily" or "full") in the filepath.
 
         Example filepath: alma-2024-03-01-daily-extracted-records-to-index_19.xml
             - run_type=daily
         """
-        return HARVEST_TYPE_MAP[self.harvest_type] in filepath
+        return [
+            filepath
+            for filepath in filepaths
+            if HARVEST_TYPE_MAP[self.harvest_type] in filepath
+        ]
+
+    def _filter_filepaths_by_dates(self, filepaths: list[str]) -> list[str]:
+        """Filter list of XML files by date.
 
-    def _filter_filepath_by_dates(self, filepath: str) -> bool:
-        """Bool if from and/or until dates align with filepath dates.
+        Given a list of XML files, the method will retrieve the date (YYYY-MM-DD)
+        from the filepath and determine whether the date is valid by checking
+        if the date falls within the specified 'from_date' and/or 'until_date'
+        arguments passed to the harvester.
 
         Example filepath: alma-2024-03-01-daily-extracted-records-to-index_19.xml
             - run_date=2024-03-01
         """
-        match = FILEPATH_DATE_REGEX.match(filepath)
-        if not match:  # pragma: nocover
-            message = f"Could not parse date from filepath: {filepath}"
-            logger.warning(message)
-            return False
+        filtered_filepaths = []
+        for filepath in filepaths:
+            filepath_date = convert_to_utc(
+                date_parser(self._get_date_from_filepath(filepath))  # type: ignore[arg-type]
```

Comment:
Don't think `# type: ignore[arg-type]` is needed.  I removed locally and passed linting!

Reply:
Thanks for the heads up!